### DISCUSSION TITLE
Free SQL IDE with DuckDB 0.10.0 integration

### DISCUSSION THREAD

**Discussion by ryanhamilton** [duckdb/duckdb#7957] [Show and tell]
We just released qStudio 2.10 - which includes DuckDB 0.8.

https://www.timestored.com/qstudio/download
https://www.timestored.com/qstudio/help/duckdb-sql-editor

Try
1. Going to File -> New DuckDB database
2. Help -> Open DuckDB Example .SQL

To quickly try out the new ASOF join. Very excited to get ASOF, also added the shorthand pivot as that notation is beautiful!

![image](https://github.com/duckdb/duckdb/assets/7273208/6703b8ab-40ef-4434-99e6-0afce554e2c7)

**Comment by l1t1** [comment]
H2 can visit duckdb too.

**Comment by mikemdr1** [comment]
DBeaver has also support for duckDB
what makes qStudio different/better?

I'm looking up for a native/simple SQL editor for duckDB (something similar like 'DB Browser' for SQLite)

**Reply by ryanhamilton** [reply]
1. It bundles duckdb as part of the application.
2. It integrates duckdb.  To create a new duckdb database you can simply go to the menu File -> New duckdb database. To open an existing duckdb database in windows you can just double-click it in file explorer.
4. DuckDB is for analytics, qStudio is for analytics, it allows intuitively charting data straight from your query. With time series charts, step plots, bar charts.....heatmaps etc.

qStudio has been used by data analysts in banks using kdb+ (another column oriented database) for over 10 years.

What qStudio doesn't do is UI query builder, diagram generators or database modification by clicking buttons. Our focus is analysis not administration.

**Comment by ryanhamilton** [comment]
qStudio 2.56 now ships with DuckDB 0.10.0

Which includes fixed length vectors and array_cross_product support so I have added them to the bundled demo code.

![image](https://github.com/duckdb/duckdb/assets/7273208/558c2f64-9c92-4420-8118-fd3947fa81f1)
